idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/DetectorMojo.java

```diff
@@ -253,18 +263,29 @@ private Void detectorExecute(final ErrorLogger logger, final MavenProject mavenP
                 }
                 Logger.getGlobal().log(Level.INFO, errorMsg);
                 logger.writeError(errorMsg);
-                return null;
+                return;
             }
         }
 
         if (this.runner == null) {
             this.runner = InstrumentingSmartRunner.fromRunner(runners.get(0), mavenProject.getBasedir());
         }
-        final List<String> tests = getOriginalOrder(mavenProject, this.runner.framework());
+    }
+
+    protected List<String> getTests(
+            final MavenProject project,
+            TestFramework testFramework) throws IOException {
+        return getOriginalOrder(project, testFramework);
+    }
+
+    protected Void detectorExecute(final ErrorLogger logger, final MavenProject mavenProject, final int rounds) throws IOException {
+        defineSettings(logger, mavenProject);
+        final List<String> tests = getTests(mavenProject, this.runner.framework());
 
         if (!tests.isEmpty()) {
             Files.createDirectories(outputPath);
             Files.write(PathManager.originalOrderPath(), String.join(System.lineSeparator(), tests).getBytes());
```

Comment:
Since the tests now come from `getTests()`, they may not be the original order anymore (since `IncDetectorMojo` allows it to be a subset), so should write out the actual original order here.